packages/ihe-gateway-sdk/src/client/ihe-gateway.ts

```diff
@@ -1,41 +1,128 @@
+   * @param xcaIti38Request[].xcpdGateway The XCA ITI-38 endpoint of the organization to make a request to.
+   * @param xcaIti38Request[].samlAttributes The SAML attributes of the org making the request.
+   * @param xcaIti38Request[].classCode Optional. The class code of the document.
+   * @param xcaIti38Request[].practiceSettingCode Optional. The practice setting code of the document.
+   * @param xcaIti38Request[].facilityTypeCode Optional. The facility type code of the document.
+   * @param xcaIti38Request[].documentCreationDate Optional. The document creation date.
+   * @param xcaIti38Request[].serviceDate Optional. The service date.
    *
-   * @throws {@link ZodError}
-   * Thrown if organization OIDs or principalCareProviderNPIs don't meet their respective criteria.
+   * @returns 202 Accepted if the request was successful.
    */
+  async getDocuments(xcaIti38Request: XCA_ITI_38Request): Promise<void> {
+    return await this.api.post(IHEGateway.XCA_ITI_38_ENDPOINT, xcaIti38Request);
+  }
+
+  /**
+   * Retrieve Documents (XCA-ITI-39) request.
+   * https://profiles.ihe.net/ITI/TF/Volume2/ITI-39.html
+   *
+   * @param xcaIti39Request An array of document retrieval transaction requests.
+   * @param xcaIti39Request[].id Unique ID for the request.
+   * @param xcaIti39Request[].cxId The customer ID.
+   * @param xcaIti39Request[].homeCommunityId The OID of the organization to make a request to.
+   * @param xcaIti39Request[].repositoryUniqueId The unique ID of the repository.
+   * @param xcaIti39Request[].documentUniqueId The unique ID of the document.
+   *
+   * @returns 202 Accepted if the request was successful.
+   */
+  async retrieveDocument(xcaIti39Request: XCA_ITI_39Request): Promise<XCPDResponse> {
+    return await this.api.post(IHEGateway.XCA_ITI_39_ENDPOINT, xcaIti39Request);
+  }
+
   createXCPDRequest({
     patient,
     cxId,
     xcpdGateways,
     principalCareProviderNPIs,
     requestId,
-  }: {
-    patient: Patient;
-    cxId: string;
-    xcpdGateways: XCPDGateways;
-    principalCareProviderNPIs?: NPIStringArray;
-    requestId?: string;
-  }): XCPDRequest {
+    org,
+  }: XCPDPayload): XCPDRequest {
+    const user = `${org.name} System User`;
+    const defaultSubjectRole = {
+      display: "Administrative AND/OR managerial worker",
+      code: "106331006",
+    };
+
     const xcpdRequest: XCPDRequest = {
-      id: requestId ?? uuidv4(), // #1263 TODO: need to change this to UUIDv7
+      id: requestId ?? uuidv4(),
       cxId,
-      xcpdGateways: xcpdGatewaysSchema.parse(xcpdGateways),
       timestamp: new Date().toISOString(),
+      xcpdGateways: xcpdGatewaysSchema.parse(xcpdGateways),
       patientResource: patient,
+      samlAttributes: {
+        subjectId: user,
+        subjectRole: defaultSubjectRole,
+        organization: org.name,
+        organizationId: org.oid,
+        homeCommunityId: org.oid,
+        purposeOfUse: "TREATMENT",
```

Comment:
Shouldn't we have this as a parameter? Could have "treatment" as a default fallback if we want to.

Reply:
This will be done in the api